dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java

```diff
@@ -112,27 +115,20 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
             String absolutePath = file.getAbsolutePath();
 
             MinimalLanguageInterface.GenericFileType fileType;
-            switch (file.getType()) {
-            case DOCKSTORE_SMK:
-            case DOCKSTORE_CWL:
-            case DOCKSTORE_WDL:
-            case NEXTFLOW_CONFIG:
-            case NEXTFLOW:
-            case DOCKSTORE_SERVICE_YML:
-            case DOCKSTORE_SERVICE_OTHER:
-            case DOCKSTORE_YML:
+
+            FileTypeCategory fileTypeCategory = file.getType().getCategory();
+            if (fileTypeCategory == FileTypeCategory.GENERIC_DESCRIPTOR
```

Comment:
1. If you do it this way, I think a switch is better, but personal preference.
2. I would be inclined to instead add a GenericFileType property to FileTypeCategory, so then you could just have `fileType = fileTypeCategory.getGenericFileType()`.

Reply:
replaced code with a switch statement